test/run

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+#
+# Test the MongoDB image.
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+
+set -exo nounset
+shopt -s nullglob
+
+. test/common-lib
+. test/lib
+
+TEST_LIST="\
+ctest_container_creation
+ctest_configuration
+ctest_general
+ctest_change_password
+ctest_mount_config
+ctest_doc_content
+ctest_WT_cache
+ctest_local_replication
+ctest_s2i"
+
+test $# -eq 1 -a "${1-}" == --list && exit 0
+test -n "${IMAGE_NAME-}" || false 'make sure $IMAGE_NAME is defined'
+
+CID_FILE_DIR=$(mktemp --suffix=mongodb_test_cidfiles -d)
+
+# enable cleanup
+ct_enable_cleanup
+
+TEST_DIR="$(readlink -zf $(dirname "${BASH_SOURCE[0]}"))"
+S2I_ARGS="--force-pull=false "
+
+function test_mongo() {
+    echo "  Testing MongoDB"
+    if [ -v ADMIN_PASS ]; then
+        echo "  Testing Admin user privileges"
+        mongo_admin_cmd "db=db.getSiblingDB('${DB}');db.dropUser('${USER}');"
+        mongo_admin_cmd "db=db.getSiblingDB('${DB}');db.createUser({user:'${USER}',pwd:'${PASS}',roles:['readWrite','userAdmin','dbAdmin']});"
+        mongo_admin_cmd "db=db.getSiblingDB('${DB}');db.testData.insert({x:0});"
+        mongo_cmd "db.createUser({user:'test_user2',pwd:'test_password2',roles:['readWrite']});"
+    fi
+    echo "  Testing user privileges"
+    mongo_cmd "db.testData.insert({ y : 1 });"
+    mongo_cmd "db.testData.insert({ z : 2 });"
+    mongo_cmd "db.testData.find().forEach(printjson);"
+    mongo_cmd "db.testData.count();"
+    mongo_cmd "db.testData.drop();"
+    mongo_cmd "db.dropDatabase();"
+    echo "  Success!"
+}
+
+function ctest_configuration() {
+    echo "  Testing image configuration settings"
+    test_config_option MONGODB_QUIET true "quiet: true"
```

Comment:
Also

    test_config_option MONGODB_PREALLOC false "preallocDataFiles: false"
    test_config_option MONGODB_NOPREALLOC true "preallocDataFiles: false"
    test_config_option MONGODB_SMALLFILES true "smallFiles: true"

was tested for `2.6`-`3.0-upg`